101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+                },
+                "parameters": {
+                  "privateDnsName": {
+                    "value": "privatelink.api.azureml.ms"
                   },
-                  "keySource": "Microsoft.Storage"
-              },
-              "supportsHttpsTrafficOnly": true
-          }
-      },
-      {
-          "condition": "[equals(parameters('keyVaultNewOrExisting'), 'new')]",
-          "type": "Microsoft.KeyVault/vaults",
-          "apiVersion": "2019-09-01",
-          "name": "[parameters('keyVaultName')]",
-          "location": "[parameters('location')]",
-          "properties": {
-              "tenantId": "[variables('tenantId')]",
-              "sku": {
-                  "name": "standard",
-                  "family": "A"
-              },
-              "accessPolicies": []
-          }
-      },
-      {
-          "condition": "[equals(parameters('applicationInsightsNewOrExisting'), 'new')]",
-          "type": "Microsoft.Insights/components",
-          "apiVersion": "2018-05-01-preview",
-          "name": "[parameters('applicationInsightsName')]",
-          "location": "[if(or(equals(parameters('location'),'eastus2'), equals(parameters('location'),'westcentralus')),'southcentralus',parameters('location'))]",
-          "kind": "web",
-          "properties": {
-              "Application_Type": "web"
-          }
-      },
-      {
-          "type": "Microsoft.MachineLearningServices/workspaces",
-          "apiVersion": "2020-03-01",
-          "name": "[parameters('workspaceName')]",
-          "location": "[parameters('location')]",
-          "dependsOn": [
-              "[variables('storageAccount')]",
-              "[variables('keyVault')]",
-              "[variables('applicationInsights')]"
-          ],
-          "identity": {
-              "type": "systemAssigned"
-          },
-          "sku": {
-              "tier": "[parameters('sku')]",
-              "name": "[parameters('sku')]"
-          },
-          "properties": {
-              "friendlyName": "[parameters('workspaceName')]",
-              "storageAccount": "[variables('storageAccount')]",
-              "keyVault": "[variables('keyVault')]",
-              "applicationInsights": "[variables('applicationInsights')]",
-              "containerRegistry": "[if(empty(parameters('containerRegistry')), json('null'), parameters('containerRegistry'))]",
-              "adbWorkspace": "[if(empty(parameters('adbWorkspace')), json('null'), parameters('adbWorkspace'))]",
-              "encryption": {
-                  "status": "[parameters('encryption_status')]",
-                  "keyVaultProperties": {
-                      "keyVaultArmId": "[parameters('cmk_keyvault')]",
-                      "keyIdentifier": "[parameters('resource_cmk_uri')]"
+                  "privateEndpointNicResourceId": {
+                    "value": "[if(equals(parameters('privateEndpointType'), 'AutoApproval'),
```

Comment:
the false side of your if() stmt is not a valid expression

Reply:
@bmoore-msft I tried. If not using this method, it will fail if the condition for the whole deployment is set to false

Reply:
if you need the if() then use a valid expression on the false side of the statement... if it's a no-op, use something that indicates that (e.g. json('null'))